R/tuneTrain.R

```diff
@@ -23,206 +27,371 @@
-        }
-        
-        else if(length(levels(data[,c(1)])) != 2){
-            x = list(Tuning = tune.mod, 
-                     Model = train.mod, 
-                     `Class Probabilities` = prob.mod, 
-                     `Class Probabilities Plot` = prob.plot, 
-                     `TrainingIndex`= Train_Index,
-                     `Training Data` = trainset, 
-                     `Test Data` = testset)
-        }
-        
-        return(x)
+    seqStop <- size + 1
+    seqInt <- 1
+    tuneGrid <- expand.grid(.size = seq(seqStart, seqStop, by=seqInt), 
+                            .decay = 0.1^(seq(0.01, 0.09, 0.01))*0.11)
+    
+    ctrl2 = caret::trainControl(method = control, number = number, 
+                                repeats = repeats, classProbs = classProbs,
+                               # sampling = subsampling,
+                                summaryFunction = summary)
+    
+    train.mod = caret::train(trainx, trainy, method, 
+                             tuneGrid = tuneGrid, tuneLength = length, 
+                             trControl = ctrl2, trace = FALSE)
+    print(train.mod)
+  }
+  else {
+    tune.mod = caret::train(trainx, trainy, method = method,
+                            tuneLength = length, trControl = ctrl)
+    print(tune.mod)
+    
+    if (method == "knn") {
+      k <- tune.mod[["bestTune"]][["k"]]
+      
+      if (k - 2 <= 0) {
+        seqStart <- k
+      }
+      else {
+        seqStart <- k - 2
+      }
+      seqStop <- k + 2
+      seqInt <- 1
+      tuneGrid <- expand.grid(.k = seq(seqStart, seqStop, by=seqInt))
     }
-}
\ No newline at end of file
+    else if (method == "rf") {
+      mtry <- tune.mod[["bestTune"]][["mtry"]]
+      
+      if (mtry - 2 <= 0) {
+        seqStart <- mtry
+      }
+      else {
+        seqStart <- mtry - 2
+      }
+      
+      seqStop <- mtry + 2
+      seqInt <- 1
+      tuneGrid <- expand.grid(.mtry = seq(seqStart, seqStop, by=seqInt))
+    }
+    else if (method == "svmLinear2") {
+      cost <- tune.mod[["bestTune"]][["cost"]]
+      
+      if (cost - 0.25 <= 0 || cost - 0.5 <= 0 || cost - 0.75 <= 0 || cost - 1 <= 0) {
+        seqStart <- cost
+      }
+      else {
+        seqStart <- cost - 1
+      }
+      seqStop <- cost + 1
+      seqInt <- 0.25
+      tuneGrid <- expand.grid(.cost = seq(seqStart, seqStop, by=seqInt))
+      
+    } 
+    
+    ctrl2 = caret::trainControl(method = control, number = number, 
+                                repeats = repeats, classProbs = classProbs, 
+                               # sampling = subsampling,
+                                summaryFunction = summary)
+    
+    train.mod = caret::train(trainx, trainy, method, 
+                             tuneGrid = tuneGrid, tuneLength = length,
+                             trControl = ctrl2)
+    print(train.mod)
+  }
+  
+  
+  ## For binary classification
+  if (is.factor(data[[y]]) && length(unique(data[[y]])) == 2) {
```

Comment:
Is the binary classification the only classification type handled in this function?